x-pack/plugins/security_solution/server/endpoint/routes/resolver.ts

```diff
@@ -7,23 +7,34 @@
 import { IRouter } from 'kibana/server';
 import { EndpointAppContext } from '../types';
 import {
-  validateTree,
+  validateTreeEntityID,
   validateEvents,
   validateChildren,
   validateAncestry,
   validateAlerts,
   validateEntities,
+  validateTree,
 } from '../../../common/endpoint/schema/resolver';
 import { handleChildren } from './resolver/children';
 import { handleAncestry } from './resolver/ancestry';
-import { handleTree } from './resolver/tree';
+import { handleTree as handleTreeEntityID } from './resolver/tree';
+import { handleTree } from './resolver/tree/handler';
 import { handleAlerts } from './resolver/alerts';
 import { handleEntities } from './resolver/entity';
 import { handleEvents } from './resolver/events';
 
 export function registerResolverRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
   const log = endpointAppContext.logFactory.get('resolver');
 
+  router.post(
```

Comment:
Once we transition the front end over to this api we'll be able to remove all the deprecated routes and their code.